osquery/tables/system/windows/users.cpp

```diff
@@ -65,42 +73,49 @@ std::string getUserHomeDir(const std::string& sid) {
   return "";
 }
 
-void processRoamingProfiles(const std::set<std::string>& processedSids,
-                            QueryData& results) {
-  QueryData regResults;
-  queryKey(kRegProfilePath, regResults);
+// If given a list of UIDs to constrain the results, check if a particular SID
+// has a RID that matches any of those UIDs.
+bool sidMatchesAnyDesiredUids(const std::set<std::string>& uidStrings,
+                              const std::string& sidString) {
+  // If there is no constraint of UIDs given, results will not be filtered
+  if (uidStrings.empty()) {
+    return true;
+  }
 
-  for (const auto& profile : regResults) {
-    Row r;
-    if (profile.at("type") != "subkey") {
-      continue;
+  auto toks = osquery::split(sidString, "-");
+  auto uid = toks.at(toks.size() - 1);
+  for (auto desiredUid : uidStrings) {
+    if (uid.compare(desiredUid) == 0) {
+      return true;
     }
+  }
 
-    auto sidString = profile.at("name");
-    if (processedSids.find(sidString) != processedSids.end()) {
-      continue;
-    }
-    r["uuid"] = sidString;
-    r["directory"] = getUserHomeDir(sidString);
+  return false;
+}
 
-    PSID sid;
-    auto ret = ConvertStringSidToSidA(sidString.c_str(), &sid);
-    if (ret == 0) {
-      VLOG(1) << "Convert SID to string failed with " << GetLastError();
-    }
+// Given a SID, retrieve information about the matching user
+void genUser(const std::string& sidString, QueryData& results) {
+  Row r;
+
+  r["uuid"] = sidString;
+  r["directory"] = getUserHomeDir(sidString);
+  r["shell"] = getUserShell(sidString);
+  r["type"] = kWellKnownSids.find(sidString) == kWellKnownSids.end()
+                  ? "roaming"
+                  : "special";
+  r["description"] = "";
```

Comment:
Why not leave this null? (nit)

Reply:
Actually now that you draw attention to it, this doesn't have to go unfilled. It can make a quick lvl 2 call to `NetUserGetInfo` and get this. I'll add that in.